### DISCUSSION TITLE
What must i do if signIn response with SignupRequired?

### DISCUSSION THREAD

**Discussion by dataflowjs** [xelaj/mtproto#64] [Q&A] [answered]
Im trying to get API access, but after SignIn, i got SignUpRequered
![1](https://user-images.githubusercontent.com/11431176/106881643-79fbfa00-66e6-11eb-8596-80cc6f7232a9.png)
But account is registered and im already logget in telegram...
What im did wrong?
I was launched auth example https://github.com/xelaj/mtproto/tree/main/examples/auth

**Comment by quenbyako** [comment, marked as answer]
@imbaMF that's a good question, cause we only tested simple sign in case, not sign up.

If i correctly understood [this](https://core.telegram.org/api/auth#sign-insign-up) part of telegram docs:
> If the code was entered correctly, but the method returns auth.authorizationSignUpRequired, it means that account with this phone number doesn't exist yet: user needs to provide basic information, accept terms of service and then the new user registration method (auth.signUp) must be invoked.

so you need to call `client.AuthSignUp` instead `client.AuthSignIn` with, i think, same arguments, plus your first name, and, optionaly, last name.

Btw, since we don't have enough free phone numbers, I will be insanely happy if you create PR with example how to create telegram account using this library. Also, you can text me [in telegram](t.me/zhopakotabot), i'll try to help you
